### PR TITLE
chore(events/api): make list APIs operate on event-level objects

### PR DESCRIPTION
**Pull request by mattwang44**
## Types of changes

- [x] **Bugfix**
- [x] **New feature**
- [x] **Refactoring**

## WHY

The schedule API returns the `id` from the event-level model in the response while the other event APIs return the `id` from the proposal-level model. It makes frontend development harder to associate detail pages to the main schedule page.

## HOW

**Major**:
- Make all APIs for retrieving data in the `events` app operates on event-level models (& return event-level id).
- Implement speech API for querying data based on given query parameters or path parameters:

```bash
# List Event API
GET /api/events/speeches/?event_types=<event types>

# examples
GET /api/events/speeches/  # list all talk, sponsored, & tutorial events
GET /api/events/speeches/?event_types=tutorial  # list all tutorial events
GET /api/events/speeches/?event_types=talk,sponsored  # list all talk & sponsored events
GET /api/events/speeches/?event_types=wrong  # return 404
```
```bash
# Detail Event API
GET /api/events/speeches/<event_type>/<pk>/

# examples
GET /api/events/speeches/talk/9  # get detail of talk event with pk=9
GET /api/events/speeches/tutorial/1  # get detail of tutorial event with pk=1
GET /api/events/speeches/wrong/2  # return 404
GET /api/events/speeches/talk/wrong  # return 404
```

**Minor**:
- refactor serializer of event API
- fix `EventWrapper` for handling `speaker_name` of sponsored event
- add a new field of `youtube_id` for talk event & sponsored event

### CODE REVIEW

File: src/events/api/views.py [hunk 1]
```diff
@@ -124,6 +132,8 @@ def title(self) -> str:
     def speakers(self) -> list:
         if isinstance(self.obj, KeynoteEvent):
             return [self.obj.speaker_name]
+        if isinstance(self.obj, SponsoredEvent):
```
Comment (setmao) [inline src/events/api/views.py:135]: Thanks for the fix.

File: src/events/api/views.py [hunk 2]
```diff
@@ -7,84 +7,92 @@
...
+            raise Http404
+
+        for _type, api_view in [
+            ('talk', TalkDetailAPIView),
+            ('sponsored', SponsoredEventDetailAPIView),
+            ('tutorial', TutorialDetailAPIView),
+        ]:
+            if _type == event_type:
+                view = api_view.as_view()
+                return view(request._request, *args, **kwargs)
```
Comment (setmao) [inline src/events/api/views.py:93]: I suggest use dict to get the API view model, and pass the request directly.
```suggestion
        api_view_dict = {
            'talk': TalkDetailAPIView,
            'sponsored': SponsoredEventDetailAPIView,
            'tutorial': TutorialDetailAPIView,
        }
        api_view = api_view_dict[event_type]
        view = api_view.as_view()
        return view(request, *args, **kwargs)
```

File: src/events/api/views.py [hunk 3]
```diff
@@ -7,84 +7,92 @@
...
+        data = []
+        for event_type, api_view in [
+            ('talk', TalkListAPIView),
+            ('sponsored', SponsoredEventListAPIView),
+            ('tutorial', TutorialListAPIView),
+        ]:
+            if event_type not in event_types or not event_types:
+                continue
+            view = api_view.as_view()
+            event_data = view(request._request, *args, **kwargs).data
```
Comment (setmao) [inline src/events/api/views.py:52]: I suggest use dict to get the API view model, and pass the request directly.
```suggestion
        api_view_dict = {
            'talk': TalkListAPIView,
            'sponsored': SponsoredEventListAPIView,
            'tutorial': TutorialListAPIView,
        }

        for event_type in event_types:
            try:
                api_view = api_view_dict[event_type]
            except KeyError:
                continue
            view = api_view.as_view()
            event_data = view(request, *args, **kwargs).data
```

**Comment by mattwang44** [comment]
@setmao 
- The suggested change is applied with modification.
- The implementation of `get_proposal_serializer_class()` is reverted.

**Review by setmao — APPROVED** [review]
LGTM！

**Pull request merged**